osu.Desktop/DiscordRichPresence.cs

```diff
@@ -114,6 +135,33 @@ private void updateStatus()
                 {
                     presence.Buttons = null;
                 }
+
+                if (!hideIdentifiableInformation && multiplayerClient.Room != null)
+                {
+                    MultiplayerRoom room = multiplayerClient.Room;
+                    presence.Party = new Party
+                    {
+                        Privacy = string.IsNullOrEmpty(room.Settings.Password) ? Party.PrivacySetting.Public : Party.PrivacySetting.Private,
+                        ID = room.RoomID.ToString(),
+                        // technically lobbies can have infinite users, but Discord needs this to be set to something.
+                        // to make party display sensible, assign a powers of two above participants count (8 at minimum).
+                        Max = (int)Math.Max(8, Math.Pow(2, Math.Ceiling(Math.Log2(room.Users.Count)))),
+                        Size = room.Users.Count,
```

Comment:
The presence must be listening to changes in number of participants here, otherwise the feature would look quite broken.

I have tried to resolve this with https://github.com/ppy/osu/pull/27443/commits/8b730acb082379174f9a48b5fd132b184b4e9a81 but it turns out the event was being triggered too much and resulting in completely unnecessary presence updates (e.g. the event triggers when the local user themselves join the room, something that `UserActivity` already takes care of).

Using `UserJoined`/`UserLeft` events *should* work, they trigger only once and trigger only when a user other than the local user joins or leaves the room.